source/service.go

```diff
@@ -122,21 +124,13 @@ func (sc *serviceSource) endpointsFromTemplate(svc *v1.Service) ([]*endpoint.End
 	}
 
 	hostname := buf.String()
-	for _, lb := range svc.Status.LoadBalancer.Ingress {
-		if lb.IP != "" {
-			//TODO(ideahitme): consider retrieving record type from resource annotation instead of empty
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.IP, ""))
-		}
-		if lb.Hostname != "" {
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.Hostname, ""))
-		}
-	}
+	endpoints = sc.generateEndpoints(svc, hostname)
 
 	return endpoints, nil
 }
 
 // endpointsFromService extracts the endpoints from a service object
-func endpointsFromService(svc *v1.Service) []*endpoint.Endpoint {
+func (sc *serviceSource) endpointsFromService(svc *v1.Service) []*endpoint.Endpoint {
```

Comment:
nitpick: since it is a method (now) do you think it makes sense to rename the function to just `endpoints()` ?